hudi-io/src/main/java/org/apache/hudi/io/hfile/HFileMetaBlock.java

```diff
@@ -56,12 +58,11 @@ public byte[] getFirstKey() {
   }
 
   @Override
-  public ByteBuffer getUncompressedBlockDataToWrite() {
-    ByteBuffer dataBuf = ByteBuffer.allocate(context.getBlockSize());
+  public ByteBuffer getUncompressedBlockDataToWrite() throws IOException {
+    ByteArrayOutputStream dataBuf = new ByteArrayOutputStream();
```

Comment:
Can we just simplify this whole method to `return ByteBuffer.wrap(entryToWrite.value);`?